### ISSUE TITLE
Add eslint.config.ts to FLAT_CONFIG_FILENAMES

### ISSUE THREAD

**Issue by f-lawe** [intlify/eslint-plugin-vue-i18n#628]
**Tell us about your environment**

- **ESLint version:** 9.23.0
- **eslint-plugin-vue version:** 10.0.0
- **eslint-plugin-vue-i18n version:** 4.0.0
- **Node version:** 22.13.1

**The problem you want to solve.**
`@intlify/vue-i18n/no-unused-keys` can't detect key usages because it can't find the ESLint config (and because of that, can't find file parsers).

**Your take on the correct solution to problem.**
If I manually add `eslint.config.ts` to `FLAT_CONFIG_FILENAMES`, everything work!

**Additional context**

**Comment by f-lawe** [comment]
@ota-meshi I saw your tags, I'll try to open a PR in the next few days 👍🏻

**Comment by f-lawe** [comment]
Oh, yes, but that's a very dirty function, called in my `eslint.config.ts`:
```ts
export const patchVueI18n = (cwd = '.') => {
  const path = cwd
    + '/node_modules/@intlify/eslint-plugin-vue-i18n'
    + '/dist/utils/parser-config-resolver/should-use-flat-config.js';

  try {
    const data = fs.readFileSync(path, 'utf8');
    const content = data.replace('\'eslint.config.js\'', '\'eslint.config.ts\'');
    fs.writeFileSync(path, content);
  }
  catch (err) {
    console.log(err);
  }
};
```

**Comment by ota-meshi** [comment]
Thanks!
